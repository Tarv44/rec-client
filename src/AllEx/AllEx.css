.date-created {
    font-style: italic;
    font-size: 12px;
}

.all-exchanges {
    min-height: 50vh;
}