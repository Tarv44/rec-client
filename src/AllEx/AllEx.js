import { NavLink } from 'react-router-dom';
import moment from 'moment'
import './AllEx.css'

export default function AllEx(props) {
    const exLis = props.exchanges.map(ex => {
        return (
            <li key={ex.id}>
                <NavLink to={`/exchange/${ex.id}`}>
                    {ex.title} <span className='date-created'>{moment(ex.date_created).format('l')}</span>
                </NavLink>
            </li>
        ) 
    })

    const exDisplay = props.exchanges.length > 0 ? <ul>{exLis}</ul> : <h5>No exchanges to display.</h5>
    return (
        <section className='all-exchanges'>
            <h2>All Exchanges</h2>
            {exDisplay}
        </section>
    )
}