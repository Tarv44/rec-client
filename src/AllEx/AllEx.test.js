import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import AllEx from './AllEx'
import store from '../dummy-store'
import { BrowserRouter } from 'react-router-dom'


describe('----- AllEx -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <AllEx exchanges={store.exchanges}/>
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <AllEx exchanges={store.exchanges}/>
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})