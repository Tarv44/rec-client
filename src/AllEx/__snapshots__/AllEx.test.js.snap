// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- AllEx ----- renders the UI as expected 1`] = `
<section
  className="all-exchanges dashboard-section"
>
  <h2>
    All Exchanges
  </h2>
  <ul>
    <li>
      <a
        aria-current={null}
        href="/exchange/1"
        onClick={[Function]}
      >
        The Roots
      </a>
      <p
        className="date-created"
      >
        2/18/2021
      </p>
    </li>
    <li>
      <a
        aria-current={null}
        href="/exchange/2"
        onClick={[Function]}
      >
        Kiwi/Aussie Rock
      </a>
      <p
        className="date-created"
      >
        2/18/2021
      </p>
    </li>
    <li>
      <a
        aria-current={null}
        href="/exchange/3"
        onClick={[Function]}
      >
        Bark bark woof
      </a>
      <p
        className="date-created"
      >
        2/18/2021
      </p>
    </li>
    <li>
      <a
        aria-current={null}
        href="/exchange/4"
        onClick={[Function]}
      >
        My influences
      </a>
      <p
        className="date-created"
      >
        2/18/2021
      </p>
    </li>
    <li>
      <a
        aria-current={null}
        href="/exchange/5"
        onClick={[Function]}
      >
        Wedding songs
      </a>
      <p
        className="date-created"
      >
        2/18/2021
      </p>
    </li>
  </ul>
</section>
`;
